presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPageSource.java

```diff
@@ -0,0 +1,243 @@
+                delegateIndexes[outputIndex] = -1;
+            }
+            else {
+                delegateIndexes[outputIndex] = delegateIndex;
+                delegateIndex++;
+            }
+            outputIndex++;
+        }
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return delegate.getCompletedBytes();
+    }
+
+    @Override
+    public long getCompletedPositions()
+    {
+        return delegate.getCompletedPositions();
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return delegate.getReadTimeNanos();
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return delegate.isFinished();
+    }
+
+    @Override
+    public Page getNextPage()
+    {
+        try {
+            Page dataPage = delegate.getNextPage();
+            if (dataPage == null) {
+                return null;
+            }
+            int batchSize = dataPage.getPositionCount();
+            Block[] blocks = new Block[prefilledBlocks.length];
+            for (int i = 0; i < prefilledBlocks.length; i++) {
+                if (prefilledBlocks[i] != null) {
+                    blocks[i] = new RunLengthEncodedBlock(prefilledBlocks[i], batchSize);
+                }
+                else {
+                    blocks[i] = dataPage.getBlock(delegateIndexes[i]);
+                }
+            }
+            return new Page(batchSize, blocks);
+        }
+        catch (RuntimeException e) {
+            closeWithSuppression(e);
+            throwIfInstanceOf(e, PrestoException.class);
+            throw new PrestoException(HUDI_CURSOR_ERROR, e);
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        try {
+            delegate.close();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public String toString()
+    {
+        return delegate.toString();
+    }
+
+    @Override
+    public long getSystemMemoryUsage()
+    {
+        return delegate.getSystemMemoryUsage();
+    }
+
+    protected void closeWithSuppression(Throwable throwable)
+    {
+        requireNonNull(throwable, "throwable is null");
+        try {
+            close();
+        }
+        catch (RuntimeException e) {
+            // Self-suppression not permitted
+            if (throwable != e) {
+                throwable.addSuppressed(e);
+            }
+        }
+    }
+
+    private static Object deserializePartitionValue(Type type, String valueString, String name, TimeZoneKey timeZoneKey)
+    {
+        if (valueString == null) {
+            return null;
+        }
+
+        try {
+            if (type.equals(BOOLEAN)) {
+                if (valueString.equalsIgnoreCase("true")) {
+                    return true;
+                }
+                if (valueString.equalsIgnoreCase("false")) {
+                    return false;
+                }
+                throw new IllegalArgumentException();
```

Comment:
Add the value here on what was illegal, so that it makes debugging easier.

Reply:
The `IllegalArgumentException` is catched later in current method, then a PrestoException with more context infomation is thrown.